src/main/java/com/iexec/resultproxy/authorization/AuthorizationService.java

```diff
@@ -136,7 +134,7 @@ public boolean checkEnclaveSignature(ResultModel model, String walletAddress) {
         boolean isSignedByEnclave = isSignedByHimself(messageHash, model.getEnclaveSignature(), enclaveChallenge);
         if (isSignedByEnclave) {
             log.info("Valid enclave signature received, allowed to push result");
-            workerpoolAuthorizations.remove(wpAuthKey);
+            authorizationRepository.delete(workerpoolAuthorization);
```

Comment:
Can `.delete` throw an exception, which would break the Result Proxy workflow? Should we wrap it in a try/catch block?

Reply:
What do you think of replacing this statement with `deleteById` which does not seem to throw an exception ?

Reply:
Sure, thanks for the idea! Let's replace `delete` with this safer method.